spring-integration-core/src/main/java/org/springframework/integration/endpoint/MessageProducerSupport.java

```diff
@@ -84,6 +96,16 @@ protected void onInit() {
 	 */
 	@Override
 	protected void doStart() {
+		Assert.state(this.outputChannel != null || StringUtils.hasText(this.outputChannelName),
+				"'outputChannel' or 'outputChannelName' is required");
+		if (this.outputChannelName != null) {
+			synchronized (this) {
```

Comment:
This is redundant. We are protected with the `lifecycleLock` in the `AbstractEndpoint.start()`